src/worker.cc

```diff
@@ -580,7 +670,6 @@ void from_json(rapidjson::Document& doc, Options& options) {
   // Set the output precision for shape/geometry (polyline encoding). Defaults to polyline6
   // This also controls the input precision for encoded_polyline in height action
   // TODO - this just for OSRM compatibility at the moment but could be supported
-  options.set_shape_format(polyline6);
```

Comment:
why these changes? there are several similar ones above

Reply:
so previously the code looked like:

1. assume the default value
2. check if they requested a value in the json
3. if they did set to what the json says

Basically we had two options, it came from json or they didnt set it and we use the default. Now we have a 3rd option, it came via protobuf and was already set. So instead of making the default regarless of what came in the json we add the lines below that say, if it wasnt already set (either by json or by protobuf as input) then we set the default, that is what is happening down on line 686